From 7255dc6ce897b571e1cd23f33c6d8f417d329654 Mon Sep 17 00:00:00 2001
From: Vladimir Lomov <vladimir@theor7.ktp>
Date: Sun, 28 Feb 2021 12:34:28 +0800
Subject: [PATCH 03/30] stat.py: added empty docstrings for functions

Signed-off-by: Vladimir Lomov <vladimir@theor7.ktp>
---
 stat.py | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/stat.py b/stat.py
index 571dcaf..96b1a29 100644
--- a/stat.py
+++ b/stat.py
@@ -3,6 +3,9 @@ import os
 
 
 def info(file_text):
+    """
+    """
+
     text_list = file_text.split()
     # method = text[list.index('start') + 2]
     start = text_list[text_list.index('start') + 2]
@@ -15,6 +18,9 @@ def info(file_text):
 
 
 def collect_statistics(folder_name, output_file_name):
+    """
+    """
+
     list_files = os.listdir(folder_name)
     with open(output_file_name, 'w') as output_file:
         for file in list_files:
@@ -33,6 +39,9 @@ def collect_statistics(folder_name, output_file_name):
 
 
 def data_to_graf(folder_name):
+    """
+    """
+
     data_list = []
     list_files = os.listdir(folder_name)
 
@@ -60,6 +69,9 @@ def data_to_graf(folder_name):
 
 
 def make_graf_file(data, start, end, method, graf_type=0):
+    """
+    """
+
     # x = steps; y = err
     if graf_type == 0:
         for i in data:
@@ -125,6 +137,9 @@ def make_graf_file(data, start, end, method, graf_type=0):
 
 
 if __name__ == '__main__':
+    """
+    """
+
     # log_files_dir = 'logs_0.1_0.15'
     # output_file = '0.1_0.15.txt'
     # collect_statistics(log_files_dir, output_file)
@@ -148,4 +163,4 @@ if __name__ == '__main__':
     # print(data)
 
     make_graf_file(data, 0.1, 0.2, 'M4', 1)
- 
\ No newline at end of file
+ 
-- 
2.30.1

