From cf1fe3b738497ecf16a08a094d00b591480dec4e Mon Sep 17 00:00:00 2001
From: Vladimir Lomov <vladimir@theor7.ktp>
Date: Sun, 28 Feb 2021 12:30:16 +0800
Subject: [PATCH 01/13] methods.py: added ability to set basic parameters

  - using environment variables.

Signed-off-by: Vladimir Lomov <vladimir@theor7.ktp>
---
 methods.py | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/methods.py b/methods.py
index 536627c..29f450c 100644
--- a/methods.py
+++ b/methods.py
@@ -1,12 +1,13 @@
 import json
+import os
 from multiprocessing import Pool
 import subprocess
 import time
 
-BASE_DIR = 'main/'
-RUN_FILE = 'main'
-RUNS = 10
-CORES = 12
+BASE_DIR = os.getenv('BASE_DIR', 'main')
+RUN_FILE = os.getenv('RUN_FILE', "main")
+RUNS = os.getenv('RUNS', 10)
+CORES = os.getenv('CORES', 12)
 
 
 def run_subprocess(params):
-- 
2.30.1

