From 56786956de46c0fc03a09f8ebb500e8d2849fbc9 Mon Sep 17 00:00:00 2001
From: Vladimir Lomov <vladimir@theor7.ktp>
Date: Sun, 28 Feb 2021 12:33:04 +0800
Subject: [PATCH 02/13] Added documentation to functions

Signed-off-by: Vladimir Lomov <vladimir@theor7.ktp>
---
 methods.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/methods.py b/methods.py
index 29f450c..f968871 100644
--- a/methods.py
+++ b/methods.py
@@ -11,7 +11,7 @@ CORES = os.getenv('CORES', 12)
 
 
 def run_subprocess(params):
-    """"
+    """Run a program with given parameters.
 
     """
     # params = (start, stop, step, method, output_file)
@@ -29,7 +29,7 @@ def run_subprocess(params):
 
 
 def get_params():
-    """
+    """Set up parameters to run a program.
 
     """
 
@@ -61,7 +61,7 @@ def get_params():
 
 
 def run():
-    """
+    """Prepare all necessary to run programs in parallel.
 
     """
     list_params = get_params()
@@ -75,6 +75,9 @@ def run():
 
 
 if __name__ == '__main__':
+    """Main 'module' of the script.
+
+    """
 
     start_time = time.time()
     print('start prgrm')
-- 
2.30.1

